main/src/vocab/base_vocab.rs

```diff
@@ -154,14 +154,15 @@ pub(crate) fn register_as_special_value(
     Ok(())
 }
 
-#[derive(Debug, Clone, Deserialize)]
+#[derive(Debug, Default, Clone, Deserialize)]
 pub struct SpecialTokenMap {
     pub unk_token: String,
     pub pad_token: Option<String>,
     pub bos_token: Option<String>,
     pub sep_token: Option<String>,
     pub cls_token: Option<String>,
     pub eos_token: Option<String>,
+    #[serde(skip_deserializing)]
```

Comment:
This should probably be removed as `SpecialTokenMap` is used by all vocabs

Reply:
You're right, but structure of `special_tokens_map.json` is different for [NLLB](https://huggingface.co/facebook/nllb-200-3.3B/blob/main/special_tokens_map.json#L209) and [M2M](https://huggingface.co/facebook/m2m100_1.2B/blob/main/special_tokens_map.json#L1). I can avoid this skip using untagged deserializer, what do you think?

Reply:
Thank you for pointing that out. It is unfortunate that models on the huggingface hub do not follow a set of conventions for the special token mask. For now I believe the best way forward would be to keep the general `SpecialTokenMap` unchanged, and indeed rely on a specific deserialization method for NLLB. You could also create a `NLLBSpecialTokenMap` that could deserialize the NLLB format, and then implement `From<NLLBSpecialTokenMap>` for `SpecialTokenMap`?

Reply:
Fixed in https://github.com/npatsakula/rust-tokenizers/pull/5.